scheds/rust/scx_p2dq/src/bpf/main.bpf.c

```diff
@@ -955,6 +953,7 @@ void BPF_STRUCT_OPS(p2dq_stopping, struct task_struct *p, bool runnable)
 	} else {
 		stat_inc(P2DQ_STAT_DSQ_SAME);
 	}
+	taskc->slice_ns = task_slice_ns(p, taskc->dsq_index);
```

Comment:
Wouldn't be more "accurate" to update the task's time slice in ops.enqueue(), instead of ops.stopping()? Potentially the task may go to sleep for a while and when it's resumed dsq time slices may have been updated.

Reply:
Yeah, that's a good call. I'm planning on reworking all the time slice handling soon, but let me fix that.